Add readall to LZ4FrameFile and use it on python 3.10
https://github.com/python-lz4/python-lz4/issues/219
https://github.com/python-lz4/python-lz4/commit/55a1e2ff557662e99edb701b844ab5ec82d74f28
--- lz4/frame/__init__.py.orig
+++ lz4/frame/__init__.py
@@ -617,6 +617,17 @@ def peek(self, size=-1):
         # returns at least one byte (except at EOF)
         return self._buffer.peek(size)
 
+    def readall(self):
+        chunks = bytearray()
+
+        while True:
+            data = self.read(io.DEFAULT_BUFFER_SIZE)
+            chunks += data
+            if not data:
+                break
+
+        return bytes(chunks)
+
     def read(self, size=-1):
         """Read up to ``size`` uncompressed bytes from the file.
 
@@ -632,6 +643,9 @@ def read(self, size=-1):
 
         """
         self._check_can_read()
+
+        if size < 0 and sys.version_info >= (3, 10):
+            return self.readall()
         return self._buffer.read(size)
 
     def read1(self, size=-1):
